src/main/java/org/folio/circulation/domain/LoanService.java

```diff
@@ -0,0 +1,51 @@
+package org.folio.circulation.domain;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.policy.LoanPolicy;
+import org.folio.circulation.domain.policy.library.ClosedLibraryStrategyService;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.Result;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.support.Result.succeeded;
+
+public class LoanService {
+
+  private final ClosedLibraryStrategyService closedLibraryStrategyService;
+
+  public LoanService(Clients clients) {
+    closedLibraryStrategyService = ClosedLibraryStrategyService.using(clients,
+      DateTime.now(DateTimeZone.UTC), false);
+  }
+
+  public CompletableFuture<Result<LoanAndRelatedRecords>> handleRecall(
```

Comment:
What does this method do? 

Does it truncate the loan when there is an outstanding recall request?

Reply:
This method updates loan in case recall exists

Reply:
My understanding is that the only update that it makes is to reduce the due date when a policy states that should happen when a recall request is present

Reply:
Yes, this logic was moved from LoanRepository

Reply:
In which I propose we change the name to describe that intent. The language of the domain seems to be truncate, so I propose `truncateLoanWhenItemRecalled`.

What do you think?

Reply:
agree, changed